framework/validators/UrlValidator.php

```diff
@@ -28,7 +28,7 @@ class UrlValidator extends Validator
      * The pattern may contain a `{schemes}` token that will be replaced
      * by a regular expression which represents the [[validSchemes]].
      */
-    public $pattern = '/^{schemes}:\/\/(([A-Z0-9][A-Z0-9_-]*)(\.[A-Z0-9][A-Z0-9_-]*)+)/i';
+    public $pattern = '/^({schemes}:\\/\\/)(www\\.)?([0-9u0000-uFFFF]*\\.\\-)?([0-9u0000-uFFFF\\-]*)\\.([a-zA-Z]{2,6})?([u0000-uFFFF]{2,6})\\/?([u0000-uFFFF0-9_\\-\\?=\\.\\&])*?/i';
```

Comment:
Good evening,

I added www as URL might have www in their name. It is optional, I can
remove that, just thought that would be good addition.
Also, double slash to worked for me in PHP as ultimate regular expression
work. Previously I had troubles with no double slash.
I can create test case for original problem. That is not a problem.

Please let me know if that is good and I do the fixes asap.

Thank you!

Best Regards,
Maksud Aghayev

*Maksud Aghayev *| Full LAMP stack web application developer/PHP Web
Developer
Linkedin: *https://www.linkedin.com/in/maksudaghayev
<https://www.linkedin.com/in/maksudaghayev>*

On Sun, Jan 17, 2016 at 11:03 PM, Dmitry Naumenko <notifications@github.com>
wrote:

> In framework/validators/UrlValidator.php
> <https://github.com/yiisoft/yii2/pull/10600#discussion_r49966864>:
>
> > @@ -28,7 +28,7 @@ class UrlValidator extends Validator
> >       * The pattern may contain a `{schemes}` token that will be replaced
> >       * by a regular expression which represents the [[validSchemes]].
> >       */
> > -    public $pattern = '/^{schemes}:\/\/(([A-Z0-9][A-Z0-9_-]*)(\.[A-Z0-9][A-Z0-9_-]*)+)/i';
> > +    public $pattern = '/^({schemes}:\\/\\/)(www\\.)?([0-9u0000-uFFFF]*\\.\\-)?([0-9u0000-uFFFF\\-]*)\\.([a-zA-Z]{2,6})?([u0000-uFFFF]{2,6})\\/?([u0000-uFFFF0-9_\\-\\?=\\.\\&])*?/i';
>
> Why did you add (www\\.)? as a separated case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yiisoft/yii2/pull/10600/files#r49966864>.
>